### PR TITLE
[symantec_endpoint_security] Merge Symantec EDR Cloud into Symantec Endpoint Security

### PR DESCRIPTION
**Pull request by janvi-elastic**
Type of change
- Enhancement

## Proposed commit message
- Add incident data-stream of Symantec EDR Cloud into Symantec Endpoint Security.
- Update Table of Contents of all dashboards.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/symantec_endpoint_security directory.
- Run the following command to run tests. 
> elastic-package test 

```
--- Test results for package: symantec_endpoint_security - START ---
╭────────────────────────────┬─────────────┬───────────┬───────────┬────────┬────────────────╮
│ PACKAGE                    │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │   TIME ELAPSED │
├────────────────────────────┼─────────────┼───────────┼───────────┼────────┼────────────────┤
│ symantec_endpoint_security │ event       │ system    │ default   │ PASS   │ 2m8.694973658s │
│ symantec_endpoint_security │ incident    │ system    │ default   │ PASS   │  36.991390064s │
╰────────────────────────────┴─────────────┴───────────┴───────────┴────────┴────────────────╯
--- Test results for package: symantec_endpoint_security - END   ---
Done
--- Test results for package: symantec_endpoint_security - START ---
╭────────────────────────────┬─────────────┬───────────┬──────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE                    │ DATA STREAM │ TEST TYPE │ TEST NAME                                                │ RESULT │ TIME ELAPSED │
├────────────────────────────┼─────────────┼───────────┼──────────────────────────────────────────────────────────┼────────┼──────────────┤
│ symantec_endpoint_security │ event       │ pipeline  │ (ingest pipeline warnings test-category-application.log) │ PASS   │ 364.089605ms │
│ symantec_endpoint_security │ event       │ pipeline  │ (ingest pipeline warnings test-category-audit.log)       │ PASS   │ 306.423918ms │
│ symantec_endpoint_security │ event       │ pipeline  │ (ingest pipeline warnings test-category-diagnostic.log)  │ PASS   │ 292.880368ms │
│ symantec_endpoint_security │ event       │ pipeline  │ (ingest pipeline warnings test-category-license.log)     │ PASS   │  313.24475ms │
│ symantec_endpoint_security │ event       │ pipeline  │ (ingest pipeline warnings test-category-security.log)    │ PASS   │ 284.728507ms │
│ symantec_endpoint_security │ event       │ pipeline  │ (ingest pipeline warnings test-category-system.log)      │ PASS   │ 282.881417ms │
│ symantec_endpoint_security │ event       │ pipeline  │ test-category-application.log                            │ PASS   │ 3.661320534s │
│ symantec_endpoint_security │ event       │ pipeline  │ test-category-audit.log                                  │ PASS   │ 1.401408111s │
│ symantec_endpoint_security │ event       │ pipeline  │ test-category-diagnostic.log                             │ PASS   │ 799.963301ms │
│ symantec_endpoint_security │ event       │ pipeline  │ test-category-license.log                                │ PASS   │ 911.965757ms │
│ symantec_endpoint_security │ event       │ pipeline  │ test-category-security.log                               │ PASS   │ 8.131367082s │
│ symantec_endpoint_security │ event       │ pipeline  │ test-category-system.log                                 │ PASS   │ 7.365000171s │
│ symantec_endpoint_security │ incident    │ pipeline  │ (ingest pipeline warnings test-incident.log)             │ PASS   │  264.28448ms │
│ symantec_endpoint_security │ incident    │ pipeline  │ test-incident.log                                        │ PASS   │ 102.452417ms │
╰────────────────────────────┴─────────────┴───────────┴──────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: symantec_endpoint_security - END   ---
Done
--- Test results for package: symantec_endpoint_security - START ---
╭────────────────────────────┬─────────────┬───────────┬─────────────────────────────────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE                    │ DATA STREAM │ TEST TYPE │ TEST NAME                                                                           │ RESULT │ TIME ELAPSED │
├────────────────────────────┼─────────────┼───────────┼─────────────────────────────────────────────────────────────────────────────────────┼────────┼──────────────┤
│ symantec_endpoint_security │             │ asset     │ dashboard symantec_endpoint_security-2926c4d6-8bd4-4591-b16f-9a2fe92d9537 is loaded │ PASS   │      1.259µs │
│ symantec_endpoint_security │             │ asset     │ dashboard symantec_endpoint_security-307802ca-681c-42d5-9518-93d8cb3f9cc6 is loaded │ PASS   │        353ns │
│ symantec_endpoint_security │             │ asset     │ dashboard symantec_endpoint_security-48a8e233-f36c-4acb-99f2-c348c6781f6f is loaded │ PASS   │        368ns │
│ symantec_endpoint_security │             │ asset     │ dashboard symantec_endpoint_security-8f4d2ce2-c5f2-421f-be7b-ffde36b86a0f is loaded │ PASS   │        386ns │
│ symantec_endpoint_security │             │ asset     │ dashboard symantec_endpoint_security-a1361bd2-d3ef-4ee3-8dfa-4930bb91be4a is loaded │ PASS   │        901ns │
│ symantec_endpoint_security │             │ asset     │ dashboard symantec_endpoint_security-bfbe8f7e-74af-4f0e-9037-793baa3709d5 is loaded │ PASS   │        390ns │
│ symantec_endpoint_security │             │ asset     │ dashboard symantec_endpoint_security-ca452d05-b541-49e3-aa31-61d619188235 is loaded │ PASS   │        399ns │
│ symantec_endpoint_security │             │ asset     │ search symantec_endpoint_security-fcfc6277-7973-4116-8358-a20c3d8f7663 is loaded    │ PASS   │        462ns │
│ symantec_endpoint_security │ event       │ asset     │ index_template logs-symantec_endpoint_security.event is loaded                      │ PASS   │        383ns │
│ symantec_endpoint_security │ event       │ asset     │ ingest_pipeline logs-symantec_endpoint_security.event-0.3.0 is loaded               │ PASS   │        186ns │
│ symantec_endpoint_security │ incident    │ asset     │ index_template logs-symantec_endpoint_security.incident is loaded                   │ PASS   │        403ns │
│ symantec_endpoint_security │ incident    │ asset     │ ingest_pipeline logs-symantec_endpoint_security.incident-0.3.0 is loaded            │ PASS   │      1.157µs │
╰────────────────────────────┴─────────────┴───────────┴─────────────────────────────────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: symantec_endpoint_security - END   ---
Done
Run static tests for the package
--- Test results for package: symantec_endpoint_security - START ---
╭────────────────────────────┬─────────────┬───────────┬──────────────────────────┬────────┬──────────────╮
│ PACKAGE                    │ DATA STREAM │ TEST TYPE │ TEST NAME                │ RESULT │ TIME ELAPSED │
├────────────────────────────┼─────────────┼───────────┼──────────────────────────┼────────┼──────────────┤
│ symantec_endpoint_security │ event       │ static    │ Verify sample_event.json │ PASS   │ 649.387377ms │
│ symantec_endpoint_security │ incident    │ static    │ Verify sample_event.json │ PASS   │  127.00076ms │
╰────────────────────────────┴─────────────┴───────────┴──────────────────────────┴────────┴──────────────╯
--- Test results for package: symantec_endpoint_security - END   ---
Done
```
## Related issues

- Closes https://github.com/elastic/integrations/issues/10212

### CODE REVIEW

File: packages/symantec_endpoint_security/data_stream/incident/agent/stream/cel.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                  string(resp.Body)
+                :
+                  string(resp.Status) + ' (' + string(resp.StatusCode) + ')'
+              ),
+            },
+          },
+          "want_more": false,
+        }
+    )
+  )
```
Comment (efd6) [inline packages/symantec_endpoint_security/data_stream/incident/agent/stream/cel.yml.hbs:103]:
```suggestion
  (
    state.want_more ?
      state
    :
      state.with({
        "limit": state.limit,
        "start_date": state.?cursor.last_timestamp.orValue(string(now - duration(state.initial_interval))),
        "end_date": now,
        "next": state.next,
      })
  ).as(state,
    post_request(
      state.url.trim_right("/") + "/v1/incidents",
      "application/json",
      {
        "limit": state.limit,
        "start_date": state.start_date,
        "end_date": state.end_date,
        "next": state.next
      }.encode_json()
    ).do_request().as(resp, resp.StatusCode == 200 ?
        bytes(resp.Body).decode_json().as(body, 
          (body.?next.orValue(body.total) != body.total).as(want_more, {
            "events": body.incidents.map(e, {
              "message": e.encode_json(),
            }),
            "next": want_more ? body.next : 0,
            "want_more": want_more,
            "limit": state.limit,
            "start_date": string(state.start_date),
            "end_date": string(state.end_date),
            "cursor": {
              ?"last_timestamp": want_more ?
                  state.?cursor.last_timestamp
                :
                  optional.of(state.end_date),
            }
          })
        )
      :
        {
          "events": {
            "error": {
              "code": string(resp.StatusCode),
              "id": string(resp.Status),
              "message": "POST:"+(
                size(resp.Body) != 0 ?
                  string(resp.Body)
                :
                  string(resp.Status) + ' (' + string(resp.StatusCode) + ')'
              ),
            },
          },
          "want_more": false,
        }
    )
  )
```
(tested against the tests here, but please check)

File: packages/symantec_endpoint_security/data_stream/incident/fields/fields.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      type: keyword
+      description: Event type category.
+    - name: conclusion
+      type: keyword
+    - name: created
+      type: date
+      description: The creation time of the incident in ISO 8601 format.
+    - name: customer_uid
+      type: keyword
+      description: Customer id.
```
Comment (efd6) [inline packages/symantec_endpoint_security/data_stream/incident/fields/fields.yml:16]:
```suggestion
      description: Customer ID.
```
(`s/id/ID/gi` please)

File: packages/symantec_endpoint_security/data_stream/incident/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      type: keyword
+    - name: id
+      type: keyword
+      description: The outcome of the event.
+    - name: incident_uid
+      type: keyword
+      description: A unique identifier for this incident.
+    - name: incident_url
+      type: keyword
+      description: The url pointing to ICDM console for this incident details.
```
Comment (efd6) [inline packages/symantec_endpoint_security/data_stream/incident/fields/fields.yml:40]:
```suggestion
      description: The URL pointing to ICDM console for this incident details.
```

File: packages/symantec_endpoint_security/img/ses-security-event.png
Comment (efd6) [inline packages/symantec_endpoint_security/img/ses-security-event.png:1]: Is the change in the complexity of the image — the number of categories — just an artifact of how the testing was done?

File: packages/symantec_endpoint_security/data_stream/incident/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -0,0 +1,470 @@
...
+          'targeted attack'
+        ].contains(ctx.ses.incident.conclusion.toLowerCase())
+  - append:
+      field: event.category
+      tag: append_intrusion_detection_into_event_category
+      value: intrusion_detection
+      if: >-
+        ctx.ses?.incident?.conclusion != null && [
+          'breach'
+        ].contains(ctx.ses.incident.conclusion.toLowerCase())
```
Comment (efd6) [inline packages/symantec_endpoint_security/data_stream/incident/elasticsearch/ingest_pipeline/default.yml:67]: If we're not expecting more categories here...
```suggestion
      if: ctx.ses?.incident?.conclusion?.equalsIgnoreCase('breach') == true
```

File: packages/symantec_endpoint_security/data_stream/incident/elasticsearch/ingest_pipeline/default.yml [hunk 5]
```diff
@@ -0,0 +1,470 @@
...
+  - append:
+      field: event.category
+      tag: append_network_into_event_category
+      value: network
+      if: >-
```
Comment (efd6) [inline packages/symantec_endpoint_security/data_stream/incident/elasticsearch/ingest_pipeline/default.yml:72]: Similar here.

File: packages/symantec_endpoint_security/data_stream/event/sample_event.json [hunk 6]
```diff
@@ -178,10 +192,19 @@
...
+            "STIC_IP_Hash",
             "HFDajsdf3254345436",
             "Cybox-Files-MD5 2",
             "Cybox-Files-SHA1 1",
-            "Cybox-Files-SHA1 2"
+            "Cybox-Files-SHA1 2",
+            "Cybox-Files-SHA2 1",
+            "Cybox-Files-SHA2 2",
+            "Cybox-Files-Parent_SHA2 1",
+            "Cybox-Files-Parent_SHA2 2"
```
[the remark this answers is not in the document]
Comment (janvi-elastic) [reply packages/symantec_endpoint_security/data_stream/event/sample_event.json:203]: Yes

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**